backed/fapi: drop Esys_TR_GetTpmHandle

williamcroberts opened the pull request:
Since the internals of the code were changed to use ESYS_TRs fully for primary
objects, to get name checking, thier was no longer a need for calls to
convert ESYS_TR's to TPM2_HANDLE's. Thus we can drop
Esys_TR_GetTpmHandle. However, the pid value is now currently unused by
the FAPI backend, so perhaps some population values should go there?
Perhaps just the ESYS_TR?

Signed-off-by: William Roberts <william.c.roberts@intel.com>

AndreasFuchsTPM reviewed (approved):
LGTM.

Regarding the pid, I have no idea. I guess that would work.

Commit:
backed/fapi: drop Esys_TR_GetTpmHandle

Since the internals of the code were changed to use ESYS_TRs fully for primary
objects, to get name checking, thier was no longer a need for calls to
convert ESYS_TR's to TPM2_HANDLE's. Thus we can drop
Esys_TR_GetTpmHandle.

Populate the pid paramter with the last 4 octets of the objects name
for debugging multiple tokens. This way all tobj->pid's are not 0.
We use the last 4, as TPM2B_NAMES all start with the name alg and
thus reduce uniqueness by two bytes.

Example Name:
000b8897499154f123ddf9fa05ff7160882f52942baba9949696d151ddd31b1e470b

We don't want the 000b aka TPM2B_ALG_SHA256.

With TPM2_PKCS11_LOGLEVEL=2, you'll observe PID being set:
pid set to: 0xAFE18287

Signed-off-by: William Roberts <william.c.roberts@intel.com>

williamcroberts commented:
@dkobras this gets rid of the esapi requirement even further.
